### PR TITLE
Destruct the singleton Communicator via testing::Environment. 

### PR DESCRIPTION
**Pull request by wujingyue**
Fixes https://github.com/NVIDIA/Fuser/issues/2497.

My other attempt (#2503) failed. There were errors that seem to be related to repeated create/delete the singleton Communicator, which I don't understand. This PR tries to create/delete the Communicator just once, keeping the behavior as close as before.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/multidevice/communicator.cpp [hunk 1]
```diff
@@ -214,6 +214,18 @@ Communicator::Communicator(
 #endif
 }
 
+Communicator::~Communicator() {
+#ifdef NVFUSER_DISTRIBUTED
```
Comment (samnordmann) [inline csrc/multidevice/communicator.cpp:218]:
```suggestion
#if(defined NVFUSER_DISTRIBUTED && defined USE_C10D_NCCL)
```

**Review by samnordmann — APPROVED** [review]
LGTM, Thanks for looking into this!

**Comment by cowanmeg** [comment]
Does this PR supersede https://github.com/NVIDIA/Fuser/pull/2503 or are both still required?

**Comment by wujingyue** [comment]
> Does this PR supersede #2503 or are both still required?

The former, as said in the PR description. I still believe #2503 should work. It currently leads to non-deterministic bugs that I couldn't understand, so I gave it up for now.

File: csrc/multidevice/communicator.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/multidevice/communicator.cpp:218]: Good catch!

**Comment by wujingyue** [comment]
@xwang233 let me know how you like the cmake file changes.

**Comment by wujingyue** [comment]
!build

**Review by xwang233 — APPROVED** [review]

**Pull request merged**